mmpose/datasets/datasets/top_down/topdown_crowdpose_dataset.py

```diff
@@ -0,0 +1,399 @@
+        all_boxes = None
+        with open(self.bbox_file, 'r') as f:
+            all_boxes = json.load(f)
+
+        if not all_boxes:
+            raise ValueError('=> Load %s fail!' % self.bbox_file)
+
+        print(f'=> Total boxes: {len(all_boxes)}')
+
+        kpt_db = []
+        num_boxes = 0
+        for n_img in range(0, len(all_boxes)):
+            det_res = all_boxes[n_img]
+            if det_res['category_id'] != 1:
+                continue
+
+            img_name = self._image_path_from_index(det_res['image_id'])
+            box = det_res['bbox']
+            score = det_res['score']
+
+            if score < self.image_thr:
+                continue
+
+            num_boxes = num_boxes + 1
+
+            center, scale = self._box2cs(box)
+            joints_3d = np.zeros((num_joints, 3), dtype=np.float)
+            joints_3d_visible = np.ones((num_joints, 3), dtype=np.float)
+            interference_joints_3d = np.zeros((0, 3), dtype=np.float)
+            kpt_db.append({
+                'image_file': img_name,
+                'center': center,
+                'scale': scale,
+                'bbox_score': score,
+                'dataset': 'crowdpose',
+                'rotation': 0,
+                'imgnum': 0,
+                'joints_3d': joints_3d,
+                'joints_3d_visible': joints_3d_visible,
+                'interference_joints': interference_joints_3d,
+            })
+        print(f'=> Total boxes after fliter '
+              f'low score@{self.image_thr}: {num_boxes}')
+        return kpt_db
+
+    def evaluate(self, outputs, res_folder, metric='mAP', **kwargs):
+        """Evaluate crowdpose keypoint results. The pose prediction results
+        will be saved in `${res_folder}/result_keypoints.json`.
+
+        Note:
+            num_keypoints: K
+
+        Args:
+            outputs (list(preds, boxes, image_path))
+                :preds (np.ndarray[1,K,5,3]): The first dimensions is the
+                    number of candidates, second third dimensions are
+                    coordinates, score is the fourth dimension of the array.
+                :boxes (np.ndarray[1,6]): [center[0], center[1], scale[0]
+                    , scale[1],area, score]
+                :image_path (list[str]): For example, [ '/', 'i','m', 'a',
+                    'g', 'e', 's', '/', '0', '0', '0', '1', '3', '3', '.',
+                    'j', 'p', 'g']
+            res_folder (str): Path of directory to save the results.
+            metric (str): Metric to be performed. Defaults: 'mAP'.
+
+        Returns:
+            name_value (dict): Evaluation results for evaluation metric.
+        """
+        assert metric == 'mAP'
+
+        res_file = os.path.join(res_folder, 'result_keypoints.json')
+
+        kpts = defaultdict(list)
+        for preds, boxes, image_path in outputs:
+            str_image_path = ''.join(image_path)
+            image_id = int(str_image_path[-10:-4])
```

Comment:
Let's use
`image_id = int(osp.basename(osp.splitext(str_image_path)[0]))`
instead of directly slicing the str.